### PR TITLE
IBX-154: Optional alt text requirement for Image FT

### PR DESCRIPTION
**Pull request by dabrt**
| Question      | Answer |
| ------------- | --- |
| JIRA Ticket   | IBX-154 |
| Versions      | 2.5 and up |

Alternative text is optional by default, it can be set as required when adding an FT to content type.

### CODE REVIEW

**Comment by dabrt** [comment]
@adamwojs Please find 5 minutes to look at this PR

File: docs/tutorials/enterprise_beginner/1_get_a_starter_website.md [hunk 1]
```diff
@@ -51,6 +51,11 @@ Edit it to remove the Image Field that has a Content Relation (ezobjectrelation)
 
 ![New image Field in the Article Content Type](img/enterprise_tut_image_in_article_ct.png)
 
+!!! note "Alternative text requirement"
+
+    In some cases you might want to require an alternative text for the image. 
+    To do this, select the "Alternative text is required" checkbox.
+
```
Comment (adamwojs) [inline docs/tutorials/enterprise_beginner/1_get_a_starter_website.md:58]: Selecting alternative text as mandatory should be our recommendation here. It's usually (and by default) used as `alt` attribute value and it's its value must not be empty. 

Ref. https://html.spec.whatwg.org/multipage/images.html#general-guidelines

**Review by adamwojs — APPROVED** [review]

File: docs/api/field_types_reference/imagefield.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ The `value` property of an Image Field returns an `\eZ\Publish\Core\FieldType\I
 |Property|Type|Example|Description|
 |------|------|------|------|
 |`id`|string|`0/8/4/1/1480-1-eng-GB/image.png`|The image's unique identifier. Usually the path, or a part of the path. To get the full path, use the `uri` property.|
-|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties.|
+|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties. This property is optional. You can require the alternative text for an image when you add the Image Field to a Content Type, by toggling the "Alternative text is required" checkbox.|
```
Comment (dabrt) [inline docs/api/field_types_reference/imagefield.md:22]:
```suggestion
|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties. This property is optional. It is recommended that you require the alternative text for an image when you add the Image Field to a Content Type, by toggling the "Alternative text is required" checkbox.|
```

File: docs/tutorials/enterprise_beginner/1_get_a_starter_website.md [hunk 3]
```diff
@@ -51,6 +51,11 @@ Edit it to remove the Image Field that has a Content Relation (ezobjectrelation)
 
 ![New image Field in the Article Content Type](img/enterprise_tut_image_in_article_ct.png)
 
+!!! note "Alternative text requirement"
+
+    In some cases you might want to require an alternative text for the image.
```
Comment (dabrt) [inline docs/tutorials/enterprise_beginner/1_get_a_starter_website.md:56]:
```suggestion
    It is recommended that you require an alternative text for the image. 
```

File: docs/api/field_types_reference/imagefield.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ The `value` property of an Image Field returns an `\eZ\Publish\Core\FieldType\I
 |Property|Type|Example|Description|
 |------|------|------|------|
 |`id`|string|`0/8/4/1/1480-1-eng-GB/image.png`|The image's unique identifier. Usually the path, or a part of the path. To get the full path, use the `uri` property.|
-|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties.|
+|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties. This property is optional. It is recommended that you require the alternative text for an image when you add the Image Field to a Content Type, by toggling the "Alternative text is required" checkbox.|
```
Comment (juskora) [inline docs/api/field_types_reference/imagefield.md:22]:
```suggestion
|`alternativeText`|string|`Picture of an apple.`|The alternative text, as entered in the Field's properties. This property is optional. It is recommended that you require the alternative text for an image when you add the Image Field to a Content Type, by selecting the "Alternative text is required" checkbox.|
```

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

**Pull request merged**